src/targets/gpu/mlir.cpp

```diff
@@ -1081,12 +1139,21 @@ tuning_config get_tuning_config_mlir(const context& migraphx_ctx,
                                      bool exhaustive)
 {
     adjust_param_shapes(m, inputs);
-
+    rewrite_reduce(m);
```

Comment:
We should probably make one function(like `mlir_rewrites`) that does `adjust_param_shapes` and `rewrite_reduce` together. This we have once place to add any additional rewrites in the future.

Reply:
I can do that after this PR.

Reply:
https://github.com/ROCm/AMDMIGraphX/issues/3384